new_reduce/cuda/reduce.hpp

```diff
@@ -0,0 +1,183 @@
+#ifndef NEW_REDUCE_CUDA_REDUCE_HPP
+#define NEW_REDUCE_CUDA_REDUCE_HPP
+
+
+#if defined(RAJA_ENABLE_CUDA)
+
+#include <cuda.h>
+#include "RAJA/policy/cuda/MemUtils_CUDA.hpp"
+#include "../util/policy.hpp"
+
+namespace detail {
+using cuda_dim_t = dim3;
+
+// ----------------------------------------------------------------------------
+//                               BLOCK REDUCE
+// ----------------------------------------------------------------------------
+  //! reduce values in block into thread 0
+  template <typename Combiner, typename T>
+  RAJA_DEVICE RAJA_INLINE T block_reduce(T val, T identity)
```

Comment:
Do we need to duplicate these functions, maybe they can be pulled into a common reduction details header?

Reply:
This duplication was a part of the prototype code and will be deleted